README.md

```diff
@@ -197,6 +202,23 @@ For video and audio: `createOfflineAudioVideoStreamInfoProviderWithCodecs()`
 
 The 2 APIs are available in [this](https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/412aab82c99a72f9dbde975f5fea81ffdc844ae5/src/include/com/amazonaws/kinesis/video/cproducer/Include.h) header file.
 
+### KVS Endpoints and DNS resolution
+The default endpoints and DNS resolution chain is implemented by the SDK. It sequentially checks each place where you can set the configuration for these parameters, and then selects the first one you set. The predefined sequence is as follows:
+
+#### Endpoint Configuration
+1. The `controlPlaneUrl` parameter for `createAbstractDefaultCallbacksProvider`.
+2. Endpoint configuration CMake parameters: (`-DAWS_KVS_USE_LEGACY_ENDPOINT_ONLY=TRUE`, `-DAWS_KVS_USE_DUAL_STACK_ENDPOINT_ONLY=TRUE`)
+3. Environment variables: (`export AWS_USE_DUALSTACK_ENDPOINT=TRUE`)
+  - If `AWS_USE_DUALSTACK_ENDPOINT` is `TRUE` (case-insensitive), the dual-stack endpoint will be used.
+4. Otherwise, the legacy endpoint will be constructed.
+
+With 2, 3, and 4, the endpoint will be constructed based on the region provided to `createAbstractDefaultCallbacksProvider`.
+
+#### DNS filtering
+1. DNS resolution CMake parameters: (`-DAWS_KVS_IPV4_ONLY=TRUE`, `-DAWS_KVS_IPV6_ONLY=TRUE`, `-DAWS_KVS_DUAL_STACK_ONLY=TRUE`)
```

Comment:
suggest adding a note - "ONLY" in this context refers to the endpoint being enabled strictly or forced.

Reply:
I didn't want to use "forced" in the readme.

I think I might rename `AWS_KVS_DUAL_STACK_ONLY` to `AWS_KVS_IPV4_OR_IPV6_ONLY`.